app/controllers/genres_controller.rb

```diff
@@ -42,6 +42,10 @@ def destroy_empty
     Genre.left_outer_joins(:tracks).where(tracks: { id: nil }).destroy_all
   end
 
+  def merge
+    render json: @genre.errors, status: :unprocessable_entity unless @genre.merge(Genre.find(params[:other_genre_id]))
```

Comment:
I think we should just say (like for tracks) that the `merge` method in `Genre` shouldn't be able to fail (and test that extensively). Because now this action works in the opposite way as it does in the tracks controller, which doesn't seem very consistent.

Rendering the updated genre (like in the tracks controller) would also be good. Just for consistency of course, since the model won't actually be changed.

Reply:
I've made the action consistent with the way this is done in tracks